src/atomate2/common/jobs/gruneisen.py

```diff
@@ -0,0 +1,250 @@
+        )
+        gru.set_mesh(
+            mesh=kpoint.kpts[0],
+            shift=compute_gruneisen_param_kwargs.get("shift", None),
+            is_gamma_center=compute_gruneisen_param_kwargs.get("is_gamma_center", True),
+            is_time_reversal=compute_gruneisen_param_kwargs.get(
+                "is_time_reversal", True
+            ),
+            is_mesh_symmetry=compute_gruneisen_param_kwargs.get(
+                "is_mesh_symmetry", True
+            ),
+        )
+    gruneisen_mesh_yaml = compute_gruneisen_param_kwargs.get(
+        "filename_mesh_yaml", "gruneisen_mesh.yaml"
+    )
+    gru._mesh.write_yaml(filename=gruneisen_mesh_yaml)  # noqa: SLF001
+    gruneisen_parameter = get_gruneisenparameter(
+        gruneisen_path=gruneisen_mesh_yaml, structure=structure
+    )
+    gp_plot = GruneisenPlotter(gruneisen=gruneisen_parameter)
+    gruneisen_mesh_plot = compute_gruneisen_param_kwargs.get(
+        "gruneisen_mesh", "gruneisen_mesh.pdf"
+    )
+    gp_plot.save_plot(
+        filename=gruneisen_mesh_plot,
+        units=compute_gruneisen_param_kwargs.get("units", "thz"),
+        img_format=compute_gruneisen_param_kwargs.get("img_format", "pdf"),
+    )
+    # get phonon band structure
+    kpath_dict, kpath_concrete = PhononBSDOSDoc.get_kpath(
+        structure=structure,
+        kpath_scheme=kpath_scheme,
+        symprec=symprec,
+    )
+
+    qpoints, connections = get_band_qpoints_and_path_connections(
+        kpath_concrete, npoints=compute_gruneisen_param_kwargs.get("npoints_band", 101)
+    )
+
+    gruneisen_band_yaml = compute_gruneisen_param_kwargs.get(
+        "filename_band_yaml", "gruneisen_band.yaml"
+    )
+    gru.set_band_structure(bands=qpoints)
+    gru._band_structure.write_yaml(filename=gruneisen_band_yaml)  # noqa: SLF001
+    gruneisen_band_structure = get_gruneisen_ph_bs_symm_line(
+        gruneisen_path=gruneisen_band_yaml,
+        structure=structure,
+        labels_dict=kpath_dict,
+    )
+    gp_bs_plot = GruneisenPhononBSPlotter(bs=gruneisen_band_structure)
+
+    GruneisenParameterDocument.get_gruneisen_weighted_bandstructure(
+        gruneisen_band_symline_plotter=gp_bs_plot,
+        save_fig=True,
+        **compute_gruneisen_param_kwargs,
+    )
+
+    gruneisen_parameter_inputs = {
+        "ground": phonopy_yaml_paths_dict["ground"],
+        "plus": phonopy_yaml_paths_dict["plus"],
+        "minus": phonopy_yaml_paths_dict["minus"],
+    }
+
+    try:
+        average_gruneisen = gruneisen_parameter.average_gruneisen()
+    except ValueError:
+        average_gruneisen = None
+
+    try:
+        thermal_conductivity_slack = gruneisen_parameter.thermal_conductivity_slack()
+    except ValueError:
+        thermal_conductivity_slack = None
+
+    derived_properties = {
+        "average_gruneisen": average_gruneisen,
+        "thermal_conductivity_slack": thermal_conductivity_slack,
+    }
+
+    return GruneisenParameterDocument.from_structure(
```

Comment:
I feel like a lot of this function could be on the task document itself. E.g. `from_phonon_yaml` or something. And then this function is only responsible for loading the files and passing the contents on. That way, if we ever want to create a builder for this workflow it will be much easier. I was also under the impression that this is how the phonon and other documents work.

Reply:
Thanks. Yes, I agree. I have adapted this.